### PR TITLE
Add sqlite persistence tests

### PR DESCRIPTION
**Pull request by jbreiding**
**What changed?**
Adding sqlite persistence tests

**Why?**
Because validation is good

**How did you test it?**
Heh

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

**Review by yiminc — APPROVED** [review]
👍

File: common/persistence/sql/sqlplugin/sqlite/cluster_metadata.go [hunk 1]
```diff
@@ -264,6 +264,5 @@ func (mdb *db) PruneClusterMembership(
 		templatePruneStaleClusterMembership,
 		constMembershipPartition,
 		mdb.converter.ToSQLiteDateTime(filter.PruneRecordsBefore),
-		filter.MaxRecordsAffected,
```
Comment (yycptt) [inline common/persistence/sql/sqlplugin/sqlite/cluster_metadata.go:267]: Shall we create an issue or add a TODO for removing this field from the filter?

**Review by yycptt — APPROVED** [review]

File: common/persistence/sql/sqlplugin/sqlite/cluster_metadata.go (hunk 1 above)
Comment (jbreiding) [reply common/persistence/sql/sqlplugin/sqlite/cluster_metadata.go:267]: I do think we need to revisit all of these sqlplugins

this copy/pasta issue needs to be prevented with some code-gen.

Comment (jbreiding) [reply common/persistence/sql/sqlplugin/sqlite/cluster_metadata.go:267]: I should probably just do that with this PR.

@yiminc thoughts?

Comment (jbreiding) [reply common/persistence/sql/sqlplugin/sqlite/cluster_metadata.go:267]: Removed the field for the filter only for sql plugin.

File: common/backoff/retry_test.go [hunk 2]
```diff
@@ -123,21 +123,22 @@ func (s *RetrySuite) TestIsRetryableSuccess() {
 
 func (s *RetrySuite) TestIsRetryableFailure() {
 	i := 0
+	theErr := someError{}
```
Comment (jbreiding) [inline common/backoff/retry_test.go:126]: This test consistently fails for me.

So adding a fix.

**Review by yiminc — APPROVED** [review]

**Pull request merged**